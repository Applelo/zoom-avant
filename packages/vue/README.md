# Zoom Avant

> Currently in _alpha_

**Zoom Avant** (pronounced /zum a.vɑ̃/) is a drilldown component for VueJS 3.

> This module is not compatible with VueJS 2

## Install

```
npm i -D @zoom-avant/vue

# yarn
yarn add -D @zoom-avant/vue

# pnpm
pnpm add -D @zoom-avant/vue
```

## Usage

```vue
<script setup lang="ts">
import { ZAvant, ZAvantMenu, ZAvantMenuItem } from '@zoom-avant/vue'
import '@zoom-avant/vue/css'
</script>

<ZAvant :dynamic-height="true">
  <ZAvantMenuItem>
    <ZAvantMenu next="Next 1">
      <ZAvantMenuItem>Hello</ZAvantMenuItem>
      <ZAvantMenuItem>world</ZAvantMenuItem>
      <ZAvantMenuItem>Bonjour</ZAvantMenuItem>
    </ZAvantMenu>
  </ZAvantMenuItem>
  <ZAvantMenuItem>
    <ZAvantMenu next="Next 2">
      <ZAvantMenuItem>Cheese</ZAvantMenuItem>
      <ZAvantMenuItem>Fromage</ZAvantMenuItem>
      <ZAvantMenuItem>Brie</ZAvantMenuItem>
      <ZAvantMenuItem>
        <ZAvantMenu>
          <template #next>
            <span>Bonjour le monde</span>
          </template>
          <template #back>
            <span>Retour</span>
          </template>
          <template #default>
            <ZAvantMenuItem>
              <a href="#">Hello</a>
            </ZAvantMenuItem>
            <ZAvantMenuItem>
              <a href="#">world</a>
            </ZAvantMenuItem>
          </template>
        </ZAvantMenu>
      </ZAvantMenuItem>
    </ZAvantMenu>
  </ZAvantMenuItem>
  <ZAvantMenuItem>
    <a href="#">Bonjour le monde</a>
  </ZAvantMenuItem>
</ZAvant>
```
