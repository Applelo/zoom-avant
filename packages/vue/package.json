{
  "name": "@zoom-avant/vue",
  "type": "module",
  "version": "0.5.0",
  "description": "",
  "main": "./dist/zoom-avant.umd.cjs",
  "module": "./dist/zoom-avant.js",
  "exports": {
    ".": {
      "import": "./dist/zoom-avant.js",
      "require": "./dist/zoom-avant.umd.cjs"
    }
  },
  "files": [
    "dist"
  ],
  "types": "dist",
  "scripts": {
    "dev": "vite build --watch",
    "build": "vite build && vue-tsc --emitDeclarationOnly --declaration",
    "preview": "vite preview"
  },
  "keywords": [],
  "author": "",
  "license": "GPL-3.0",
  "dependencies": {
    "vue": "^3.2.41",
    "@vueuse/core": "^9.3.1"
  },
  "devDependencies": {
    "@types/lodash.get": "^4.4.7",
    "@vitejs/plugin-vue": "^3.1.2",
    "typescript": "^4.8.4",
    "vite": "^3.1.8",
    "vite-plugin-dts": "^1.6.6",
    "lodash.get": "^4.4.2",
    "vue-tsc": "^1.0.8"
  }
}
