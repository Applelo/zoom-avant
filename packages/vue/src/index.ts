import.meta.glob('./assets/css/*.css', { eager: true })

import ZAvant from './components/ZAvant.vue'
import ZAvantMenu from './components/ZAvantMenu.vue'
import ZAvantMenuItem from './components/ZAvantMenuItem.vue'

export { ZAvant, ZAvantMenu, ZAvantMenuItem }
