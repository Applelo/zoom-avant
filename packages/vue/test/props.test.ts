import { describe, it, expect } from 'vitest'
import { defineComponent } from 'vue'
import { render } from '@testing-library/vue'
import { ZAvantPlugin } from '..'

describe('props root', () => {
  const Root = defineComponent({
    template: `
    <ZAvant :dynamic-height="true" back="Retour">
      <ZAvantItem>
        <ZAvantMenu next="go-level-1">
          <ZAvantItem>1</ZAvantItem>
          <ZAvantItem>2</ZAvantItem>
          <ZAvantItem>
            <ZAvantMenu next="go-level-2">
              <ZAvantItem>
                1
              </ZAvantItem>
              <ZAvantItem>
                2
              </ZAvantItem>
            </ZAvantMenu>
          </ZAvantItem>
        </ZAvantMenu>
      </ZAvantItem>
      <ZAvantItem>
        1
      </ZAvantItem>
    </ZAvant>`
  })

  render(Root, {
    global: {
      plugins: [ZAvantPlugin]
    }
  })
  const zavantRoot = document.querySelector<HTMLElement>('.zavant')

  it('global back', () => {
    expect(zavantRoot?.querySelectorAll('.zavant__back').length).toBe(2)
  })

  it('dynamicHeight', () => {
    expect(
      zavantRoot?.classList.contains('zavant--dynamic-height')
    ).toBeTruthy()
  })
})

describe('props root', () => {
  const Root = defineComponent({
    template: `
    <ZAvant>
      <ZAvantItem>
        <ZAvantMenu next="Suivant" back="Retour">
          <ZAvantItem>
            1
          </ZAvantItem>
        </ZAvantMenu>
      </ZAvantItem>
      <ZAvantItem>
        1
      </ZAvantItem>
    </ZAvant>`
  })

  render(Root, {
    global: {
      plugins: [ZAvantPlugin]
    }
  })
  const zavantRoot = document.querySelector<HTMLElement>('.zavant')

  it('next and back', () => {
    expect(zavantRoot?.outerHTML).toMatchSnapshot()
  })
})
