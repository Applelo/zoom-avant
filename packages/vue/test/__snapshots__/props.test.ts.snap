// Vitest Snapshot v1

exports[`props menu > next and back 1`] = `
"<nav class=\\"zavant zavant--level-0\\">
  <ul class=\\"zavant__menu zavant__menu--wrapper\\" role=\\"menubar\\" style=\\"transform: translateX(0%);\\" aria-multiselectable=\\"false\\" aria-orientation=\\"vertical\\">
    <li class=\\"zavant__item\\" role=\\"none\\"><button class=\\"zavant__next\\" role=\\"menuitem\\" aria-haspopup=\\"true\\" aria-expanded=\\"false\\" aria-controls=\\"zavant-menu-16\\">Suivant</button>
      <ul id=\\"zavant-menu-16\\" class=\\"zavant__menu\\" role=\\"menu\\">
        <li role=\\"none\\" class=\\"zavant__item zavant__item--back\\"><button class=\\"zavant__back\\" role=\\"menuitem\\">Retour</button></li>
        <li class=\\"zavant__item\\" role=\\"none\\"> 1 </li>
      </ul>
    </li>
    <li class=\\"zavant__item\\" role=\\"none\\"> 1 </li>
  </ul>
</nav>"
`;
